Content.Server/Botany/Components/PlantAnalyzerComponent.cs

```diff
@@ -0,0 +1,33 @@
+using Content.Shared.DoAfter;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Botany.Components;
+
+/// <summary>
+///    After scanning, retrieves the target Uid to use with its related UI.
+/// </summary>
+[RegisterComponent]
+public sealed partial class PlantAnalyzerComponent : Component
+{
+    [DataDefinition]
+    public partial struct PlantAnalyzerSetting
+    {
```

Comment:
Is there a reason why you put some fields in this struct instead of directly on the comp class? Also why is `ScanningEndSound` not a "setting"?

Also, please put a xmldoc summary on what `AdvancedScan` does (the rest are self-explanatory by the field names)

Reply:
I was told to use a struct, with no explanation why, so I guessed it has something to do with making it easier to change the scan delay values in the .yml instead of having to compile the component file again.